src/transactions/feed/EarnFeedItem.tsx

```diff
@@ -0,0 +1,157 @@
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import TokenDisplay from 'src/components/TokenDisplay'
+import Touchable from 'src/components/Touchable'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { getDynamicConfigParams } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs } from 'src/statsig/types'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import TransactionFeedItemImage from 'src/transactions/feed/TransactionFeedItemImage'
+import {
+  EarnClaimReward,
+  EarnDeposit,
+  EarnWithdraw,
+  TransactionStatus,
+} from 'src/transactions/types'
+
+interface AmountDisplayProps {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+  isLocal: boolean
+}
+
+function AmountDisplay({ transaction, isLocal }: AmountDisplayProps) {
+  let amountValue
+  let tokenId
+
+  switch (transaction.__typename) {
+    case 'EarnDeposit':
+      amountValue = new BigNumber(-transaction.inAmount.value)
+      tokenId = transaction.outAmount.tokenId
+      break
+    case 'EarnWithdraw':
+      amountValue = new BigNumber(transaction.outAmount.value)
+      tokenId = transaction.inAmount.tokenId
+      break
+    case 'EarnClaimReward':
+      amountValue = new BigNumber(transaction.amount.value)
+      tokenId = transaction.amount.tokenId
+      break
+  }
+
+  const textStyle = isLocal
+    ? styles.amountSubtitle
+    : [styles.amountTitle, transaction.__typename !== 'EarnDeposit' && { color: Colors.primary }]
+
+  return (
+    <TokenDisplay
+      amount={amountValue}
+      tokenId={tokenId}
+      showLocalAmount={isLocal}
+      showSymbol={true}
+      showExplicitPositiveSign={!isLocal}
+      hideSign={!!isLocal}
+      style={textStyle}
+      testID={`EarnFeedItem/${transaction.__typename}-amount-${isLocal ? 'local' : 'crypto'}`}
+    />
+  )
+}
+
+interface AmountProps {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+}
+
+function Amount({ transaction }: AmountProps) {
+  return (
+    <View style={styles.amountContainer}>
+      <AmountDisplay transaction={transaction} isLocal={false} />
+      <AmountDisplay transaction={transaction} isLocal={true} />
+    </View>
+  )
+}
+
+interface Props {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+}
+
+export default function EarnFeedItem({ transaction }: Props) {
```

Comment:
Why not make 3 separate components (deposit, withdraw, reward claim)?

Reply:
We could, I just thought this would lead to a lot of duplicate code across the components.

Reply:
I guess these are similar enough to be in a single component, but not a fan of the multiple conditional rendering inside title and subtitle, how about something like
```
const TX_TYPE_TO_STRINGS = {
  'EarnDeposit': {
    title: t('earnFlow.transactionFeed.earnDepositTitle'),
    subtitle: t('...'),
  }
  ..
}
```
and then you can do `TX_TYPE_TO_STRINGS[transaction.__typename].title / subtitle` in the JSX
or you can also do a switch case similar to AmountDisplay

Reply:
Used a switch in be4243dc52188876881cc2b93cc82d22c15e42bc!